lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java

```diff
@@ -440,15 +442,45 @@ public Attachment newAttachment() {
         return new AttachmentImpl();
     }
 
-    private void setAttachments(TaskImpl result) {
-        List<AttachmentImpl> attachmentImpls = attachmentMapper.findAttachmentsByTaskId(result.getId());
+    private void organizeAttachments(TaskImpl taskImpl) {
         List<Attachment> attachments = new ArrayList<>();
+        List<AttachmentImpl> attachmentImpls;
+        List<ClassificationSummary> classificationSummaries;
+        List<String> classificationKeyList;
+        String[] classificationKeys;
+        String domain = taskImpl.getDomain();
+
+        attachmentImpls = attachmentMapper.findAttachmentsByTaskId(taskImpl.getId());
         if (attachmentImpls != null && !attachmentImpls.isEmpty()) {
-            for (AttachmentImpl attImpl : attachmentImpls) {
-                attachments.add(attImpl);
+            attachments.addAll(attachmentImpls);
+            classificationKeyList = new ArrayList<>();
+            for (Attachment attachment : attachments) {
+                if (!classificationKeyList.contains(attachment.getClassificationKey())) {
+                    classificationKeyList.add(attachment.getClassificationKey());
+                }
+            }
+            classificationKeys = classificationKeyList.toArray(new String[0]);
+
+            try {
+                classificationSummaries = classificationService.createClassificationQuery()
```

Comment:
wollten wir nicht mit alle Classifications (also auch die aus der Task) gleichzeitig laden? Dann wäre das Laden in dieser Methode falsch angesiedelt.

Reply:
Geändert. Finales Update kommt dann mit bernds Ticket, wenn Task mit den Summaries dann angepasst wurde.